tardis/gui.py

```diff
@@ -43,68 +105,739 @@ def __init__(self, parent=None):
+        quitAction.setIcon(QtGui.QIcon(os.path.join(self.path, 
+            'closeicon.png')))
+        quitAction.triggered.connect(self.close)
+        
+        self.viewMdv = QtGui.QAction("View &Model", self)
+        self.viewMdv.setIcon(QtGui.QIcon(os.path.join(self.path,
+            'mdvswitch.png')))
+        self.viewMdv.setCheckable(True)
+        self.viewMdv.setChecked(True)
+        self.viewMdv.setEnabled(False)
+        self.viewMdv.triggered.connect(self.switchToMdv)
+        
+        self.viewForm = QtGui.QAction("&Edit Model", self)
+        self.viewForm.setIcon(QtGui.QIcon(os.path.join(self.path,
+            'formswitch.png')))
+        self.viewForm.setCheckable(True)
+        self.viewForm.setEnabled(False)
+        self.viewForm.triggered.connect(self.switchToForm)
+
+        #Menubar
+        self.fileMenu = self.menuBar().addMenu("&File")
+        self.fileMenu.addAction(quitAction)
+        self.viewMenu = self.menuBar().addMenu("&View")
+        self.viewMenu.addAction(self.viewMdv)
+        self.viewMenu.addAction(self.viewForm)
+        self.helpMenu = self.menuBar().addMenu("&Help")
+
+        #Toolbar
+        fileToolbar = self.addToolBar("File")
+        fileToolbar.setObjectName("FileToolBar")  
+        fileToolbar.addAction(quitAction)
+
+        viewToolbar = self.addToolBar("View")
+        viewToolbar.setObjectName("ViewToolBar")
+        viewToolbar.addAction(self.viewMdv)
+        viewToolbar.addAction(self.viewForm)
+
+        #Central Widget
+        self.stackedWidget = QtGui.QStackedWidget()
+        self.mdv = ModelViewer() 
+        self.stackedWidget.addWidget(self.mdv)
+        
+        #In case of active mode
+        if self.mode == 'active':
+            #Disabled currently
+            # self.formWidget = ConfigEditor(config)
+            # #scrollarea
+            # scrollarea = QtGui.QScrollArea()
+            # scrollarea.setWidget(self.formWidget)
+            # self.stackedWidget.addWidget(scrollarea)
+            # self.viewForm.setEnabled(True)
+            # self.viewMdv.setEnabled(True)
+            # model = run_tardis(config, atom_data)
+            # self.show_model(model)
+            raise TemporarilyUnavaliable("The active mode is under development"+
+                ". Please use the passive mode for now.")
+
+        self.setCentralWidget(self.stackedWidget)
+
+    def show_model(self, model=None):
+        if model:
+            self.mdv.change_model(model)
+        if model.converged:
+            self.successLabel.setText('<font color="green">converged</font>')
+        if self.mode == 'active':
+            self.modeLabel.setText('Active Mode')
+
+        self.mdv.fillOutputLabel()
+        self.mdv.tableview.setModel(self.mdv.tablemodel)
+        self.mdv.plot_model()
+        self.mdv.plot_spectrum()
+        self.showMaximized()
+
+    #Note to self: Add slot decorator decorator. And try to get all the
+    #view changer slots into a single function
```

Comment:
Don't add personal notes. You might forget to delete them before merging. Try to use the checkboxes like I did in #272